gulp.d/tasks/build-preview-pages.js

```diff
@@ -45,7 +45,17 @@ module.exports = (src, previewSrc, previewDest, sink = () => map()) => (done) =>
             const uiModel = { ...baseUiModel }
             uiModel.page = { ...uiModel.page }
             uiModel.siteRootPath = siteRootPath
-            uiModel.uiRootPath = path.join(siteRootPath, '_')
+
+            // The following has been customized to enable pull request preview builds on Github Pages.
+            // It differs from the Antora Default UI by allowing for an optional UI_ROOT_PATH_PREFIX.
+            // This environment variable is set during build to add a prefix to the {{ uiRootPath }} variable.
+            // This enables us to have preview builds that live in their own subdirectory on GitHub Pages.
+            let uiRootPath = path.join(siteRootPath, '_')
+            if (uiModel.env.UI_ROOT_PATH_PREFIX) {
+              uiRootPath = path.join('/', uiModel.env.UI_ROOT_PATH_PREFIX, uiRootPath)
+            }
+            uiModel.uiRootPath = uiRootPath
```

Comment:
This is the only real deviation necessary from the default antora UI build task. In order to support having the preview build reside in a subdirectory, I needed a way to provide the path prefix at build time.